internal/environments/environments_test.go

```diff
@@ -0,0 +1,99 @@
+// Tests for the EnvironmentController
+
+package environments
+
+import (
+	"log"
+	"testing"
+
+	//"github.com/broadinstitute/sherlock/internal/sherlock"
+	"github.com/broadinstitute/sherlock/internal/db"
+	"github.com/broadinstitute/sherlock/internal/tools"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"gorm.io/gorm"
+)
+
+func TestCreateEnvironments(t *testing.T) {
+	testCases := []struct {
+		name                string
+		request             CreateEnvironmentRequest
+		expectedError       error
+		expectedEnvironment Environment
+	}{
+		{
+			name: "creates a valid environment",
+			request: CreateEnvironmentRequest{
+				Name: "terra-juyang-opera-fish",
+			},
+			expectedError: nil,
+			expectedEnvironment: Environment{
+				Name: "terra-juyang-opera-fish",
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			testApp := initTestApp()
+			newEnvironment, err := testApp.Environments.CreateNew(testCase.request)
+
+			assert.Equal(t, testCase.expectedEnvironment.Name, newEnvironment.Name)
+			assert.Equal(t, testCase.expectedError, err)
+		})
+	}
+}
+
+// Test Environment Setup
+var (
+	Config = viper.New()
+)
+
+type TestApplication struct {
+	Environments *EnvironmentController
+	DB           *gorm.DB
+}
+
+// New returns a new instance of the core sherlock application
+func initTestApp() *TestApplication {
```

Comment:
If you are going to write tests that expect a real postgres instance can you include [this](https://github.com/broadinstitute/sherlock/blob/72cc5d0157c3e9a246786f403ee13cc11ea7aeb6/internal/sherlock/sherlock_integration_test.go#L438) in the setup code. There are points in the ci pipeline and the make tooling that only run the unit tests (those without external dependencies) for example the image build ci job. This will enable these tests to be skipped when there isn't an actual postgres instance available.